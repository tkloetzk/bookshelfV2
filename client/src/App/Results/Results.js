import React from 'react'
import Grid from '@material-ui/core/Grid'
import PropTypes from 'prop-types'
import Book from '../Book/Book'
import sortBooklist from '../../util/calculator'

const Results = ({ booklist }) => (
  <Grid container justify="center">
    {sortBooklist(booklist).map((book) => (
      <Book key={book.isbn} book={book} />
    ))}
  </Grid>
)

Results.propTypes = {
  booklist: PropTypes.arrayOf(PropTypes.shape).isRequired,
}
export default Results
