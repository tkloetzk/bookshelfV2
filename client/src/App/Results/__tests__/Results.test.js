import { MuiThemeProvider } from '@material-ui/core/styles'
import React from 'react'
import { render } from '@testing-library/react'
import muiTheme from '../../../config/themeConfig'
import Results from '../Results'
import '@testing-library/jest-dom/extend-expect'

describe('render', () => {
  let booklist
  beforeEach(() => {
    booklist = [
      {
        categories: ['Toddler', 'Parenting'],
        read: false,
        owned: true,
        _id: '5c7893eabc9b222a3547548f',
        amazonAverageRating: 4.2,
        amazonRatingsCount: 611,
        price: '',
        isbn: '9780553381436',
        title: 'The Happiest Toddler on the Block',
        subtitle:
          'The New Way to Stop the Daily Battle of Wills and Raise a Secure and Well-Behaved One- to Four-Year-Old',
        description:
          "The pediatrician-author of The Happiest Baby on the Block offers parents a groundbreaking new approach for dealing with toddlers, drawing a parallel between a child's development and humankind's journey to civilization and presenting specific behavioral techniques designed to enhance parent-child communication, alleviate tantrums, and increase positive relationships. Reprint. 60,000 first printing.",
        thumbnail:
          'http://books.google.com/books/content?id=njQpY18IazsC&printsec=frontcover&img=1&zoom=1&source=gbs_api',
        goodreadsAverageRating: 3.5,
        goodreadsRatingsCount: 5451,
        __v: 0,
      },
    ]
  })

  describe('renders', () => {
    it('should render as expected', () => {
      const { asFragment } = render(
        <MuiThemeProvider theme={muiTheme}>
          <Results booklist={booklist} />
        </MuiThemeProvider>
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
