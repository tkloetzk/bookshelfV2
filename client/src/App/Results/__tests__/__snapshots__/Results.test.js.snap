// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders should render as expected 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-104   MuiPaper-rounded"
    >
      <div
        class="makeStyles-header-106"
      >
        <div
          class="MuiCardHeader-root makeStyles-cardHeader-109"
        >
          <div
            class="MuiCardHeader-avatar makeStyles-avatar-111"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              3.85
            </p>
          </div>
          <div
            class="MuiCardHeader-content makeStyles-headerContent-110"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter"
            >
              The Happiest Toddler on the Block
            </p>
          </div>
          <div
            class="MuiCardHeader-action makeStyles-iconButton-108"
          >
            <button
              aria-label="owned"
              class="MuiButtonBase-root MuiIconButton-root"
              data-testid="ownedIcon"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <span
          class="MuiTypography-root MuiTypography-caption MuiTypography-alignCenter"
        >
          Toddler, Parenting
        </span>
      </div>
      <img
        class="MuiCardMedia-root MuiCardMedia-media makeStyles-media-107"
        src="http://books.google.com/books/content?id=njQpY18IazsC&printsec=frontcover&img=1&zoom=1&source=gbs_api"
        title="The Happiest Toddler on the Block"
      />
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root makeStyles-description-112 MuiTypography-caption MuiTypography-colorTextSecondary"
        >
          The pediatrician-author of The Happiest Baby on the Block offers parents a groundbreaking new approach for dealing with toddlers, drawing a parallel between a child's development and humankind's journey to civilization and presenting specific behavioral techniques designed to enhance parent-child communication, alleviate tantrums, and increase positive relationships. Reprint. 60,000 first printing.
        </p>
      </div>
      <div
        class="MuiCardActions-root makeStyles-expandAction-114 MuiCardActions-spacing"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root makeStyles-expand-113"
          data-testid="expandButton"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
